engine/src/mac-dialog.mm

```diff
@@ -219,14 +219,28 @@ void MCPlatformBeginFolderDialog(MCPlatformWindowRef p_owner, MCStringRef p_titl
         [t_panel setMessage: [NSString stringWithMCStringRef: p_prompt]];
     }
     else
-        [t_panel setTitle: [NSString stringWithMCStringRef: p_prompt]];
+    {
+        extern uint4 MCmajorosversion;
+        if (MCmajorosversion >= 0x10B0)
+            [t_panel setMessage: [NSString stringWithMCStringRef: p_prompt]];
+        else
+            [t_panel setTitle: [NSString stringWithMCStringRef: p_prompt]];
+        
+    }
+        //[t_panel setTitle: [NSString stringWithMCStringRef: p_prompt]];
```

Comment:
Yes - so I think the behavior should be that 'prompt' takes precedence over 'title'. So we ensure at least the 'prompt' is shown in all cases. Could we abstract the routine here so it can be used in all cases an NSOpenPanel is used? That way we only have to maintain one place where these decisions have to be made.